text/0000-cfg-path-version.md

```diff
@@ -0,0 +1,635 @@
+just sits around. Over time, this garbage can grow to a non-trivial amount.
+
+However, if we provide LTS channels in the style of [RFC 2483],
+then there are opportunities to perform some "garbage collection"
+of definitions that won't be used when the LTS version changes.
+
+# Rationale and alternatives
+[alternatives]: #rationale-and-alternatives
+
+## `accessible(..)`
+
+The primary rationale for the `accessible` mechanism is that when you
+want to support some library feature, it is some path you are thinking of
+rather than what version it was added. For example, if you want to use
+`ManuallyDrop`, you can just ask if it exists. The `version` is instead a
+proxy for the feature. Instead of detecting if the path we want is available
+or not via an indirection, we can just check if the path exists directly.
+This way, a user does not have to look up the minimum version number for
+the feature.
+
+You may think that `version = ".."` subsumes `accessible(..)`.
+However, we argue that it does not. This is the case because at the time of
+enabling the `nightly` feature that enables the path in the standard library,
+we do not yet know what minimum version it will be supported under.
+If we try to support it with `version = ".."`, it is possible that we may
+need to update the minimum version some small number of times.
+However, doing so even once means that you will need to release new versions
+of your crate. If you instead use `accessible(..)` you won't need to use
+it even once unless the name of the path changes in-between.
+
+### Preventing relative paths
+
+The reason why we have enforced that all paths must start with `::` inside
+`accessible(..)` is that if we allow relative paths, and users write
+`accessible(self::foo)`, then they can construct situations such as:
+
+```
+#[cfg(accessible(self::bar)]
+fn foo() {}
+
+#[cfg(accessible(self::foo)]
+fn bar() {}
+```
+
+One way around this is to collect all items before `cfg`-stripping,
+but this can cause problems with respect to stage separation.
+Therefore, we prevent this from occurring with a simple syntactic check.
+
+### `#[cfg(accessible(..))` or `#[cfg(accessible = ..)`
+
+We need to decide between the syntax `accessible(..)` or `accessible = ..`.
+The reason we've opted for the former rather than the latter is that the
+former syntax looks more like a question/query whilst the latter looks more
+like a statement of fact.
+
+In addition, if we would like to enable `accessible = $path` we would need to
+extend the meta grammar. We could justify that change in and of itself by
+observing that crates such as `serde_derive` permit users to write things like
+`#[serde(default = "some::function")]`. By changing the grammar we can allow
+users to instead write: `#[serde(default = some::function)]`.
+However, in this case, `accessible($path)` seems the optimal notation.
+
+If we would like to extend the meta grammar, we could do so by changing:
+
+```abnf
```

Comment:
GitHub doesn't like `_`, `:` and `|` in the "abnf" language. Perhaps just turn the highlighting off?

Reply:
Shame :( Fixed :)